platform/external-system-api/src/com/intellij/openapi/externalSystem/model/project/ExternalModuleBuildClasspathPojo.java

```diff
@@ -3,21 +3,21 @@
 
 import org.jetbrains.annotations.NotNull;
 
-import java.util.Collections;
+import java.util.ArrayList;
 import java.util.List;
 
 /**
  * @author Vladislav.Soroka
  */
 public class ExternalModuleBuildClasspathPojo {
 
-  @NotNull private final List<String> myEntries;
   @NotNull private String myPath;
+  @NotNull private List<String> myEntries;
 
   @SuppressWarnings("UnusedDeclaration")
   public ExternalModuleBuildClasspathPojo() {
     // Used by IJ serialization
-    this("___DUMMY___", Collections.emptyList());
```

Comment:
We can, the only change that is required in this file is adding the setter and removal of final from myEntries. This is enough to allow this class to be serialized.

Reply:
Thanks.  I updated the commit.

Reply:
Just to keep you updated: we see sync taking substantially more time and memory on large projects, and we think this might be a cause as each module gets its own module class path object?